service/store/store.go

```diff
@@ -517,30 +520,20 @@ func (s *Store) dealDataFilePathFor(bidID auction.BidID, payloadCid string) stri
 }
 
 // enqueueDataURI queues a data uri fetch.
-// commitTxn will be committed internally!
-func (s *Store) enqueueDataURI(commitTxn ds.Txn, b *Bid) error {
-	// Set the bid to "fetching_data"
-	if err := s.saveAndTransitionStatus(commitTxn, b, BidStatusFetchingData); err != nil {
-		return fmt.Errorf("updating status (fetching_data): %v", err)
-	}
-	if commitTxn != nil {
-		if err := commitTxn.Commit(); err != nil {
-			return fmt.Errorf("committing txn: %v", err)
+func (s *Store) enqueueDataURI(txn ds.Txn, b *Bid) error {
+	select {
```

Comment:
removing the useless goroutine. the `select` never blocks - it has default branch. Also transit to `BidStatusFetchingData` only after the job is submitted.